### DISCUSSION TITLE
Expand abbreviation without trailing space

### DISCUSSION THREAD

**Discussion by tobiasmuehl** [olets/zsh-abbr#97] [Ideas] [answered]
`abbr s='ssh root@'`

When I type `s` and then immediately follow that by pasting in a hostname, there is a pesky space that prevents the command from working

```
ssh root@ 1.2.3.4       
ssh: connect to host  port 22: Connection refused
```

Would be nice if there was a flag on the abbreviation itself to expand without the trailing space

**Comment by olets** [comment]
Thanks for the suggestion!

I've thought about adding a flag like that, or similar flags. Enhancement of zsh-abbr is a low priority for me currently. If someone's interested in contributing the feature please say so, but know that that prioritization will also impact collaborating on design, reviewing code, etc. This particular change would be low on the list because…

Today you can get very close by binding `abbr-expand` to something. See https://zsh-abbr.olets.dev/advanced.html#widgets-and-key-bindings

For example on my machine this makes <kbd>option</kbd><kbd>space</kbd> expand abbreviations without adding a space

```shell
bindkey "^[ " abbr-expand
```

To determine the key sequence code, do <kbd>control</kbd><kbd>v</kbd> followed by the desired trigger. Note that <kbd>shift</kbd><kbd>space</kbd> is not available because it sends the same sequence as plain <kbd>space</kbd>… unless your terminal app can send custom codes, like https://superuser.com/a/740174.

Or if you always connect to the same hosts, you could `abbr s1="ssh root@1.2.3.4`, `abbr s2="ssh root@2.3.4.5`, etc.

**Comment by qadzek** [comment, marked as answer]
The latest release makes this possible. See https://zsh-abbr.olets.dev/essential-commands.html#use-cursor-markers
